### PR TITLE
Make sure we auto confirm invites when autoconfirm is enabled

### PR DESCRIPTION
**Pull request by biilmann**

### CODE REVIEW

File: api/invite.go [hunk 1]
```diff
@@ -54,19 +54,23 @@ func (a *API) Invite(w http.ResponseWriter, r *http.Request) error {
...
-		return internalServerError("Error sending confirmation mail").WithInternalError(err)
-	}
-
 	user.SetRole(config.JWT.DefaultGroupName)
 	if err = a.db.UpdateUser(user); err != nil {
 		return internalServerError("Database error updating user").WithInternalError(err)
 	}
 
+	if config.Mailer.Autoconfirm {
+		user.Confirm()
```
Comment (brycekahle) [inline api/invite.go]: This does not save to the db, only mutates the user. Thus, coming after the `UpdateUser` means those changes will be lost.

**Comment by biilmann** [comment]
Good points - had completely forgotten about the need to set a password. I think in reality what would fix the issue that triggered this is a better error message when a user tries to log in from an invite that still hasn't been confirmed...

**Comment by brycekahle** [comment]
I think the real bug is #113 which I'm fixing now.

**Comment by biilmann** [comment]
Awesome

**Pull request closed without merging**